community/kernel/src/main/java/org/neo4j/storageengine/api/schema/IndexReader.java

```diff
@@ -91,5 +93,47 @@ public boolean hasFullNumberPrecision( IndexQuery... predicates )
         {
             return true;
         }
+
+        @Override
+        public void query( IndexCursorProgressor.NodeValueCursor cursor, IndexQuery... query )
+        {
+            cursor.done();
+        }
+
+        @Override
+        public void scan( IndexCursorProgressor.NodeValueCursor cursor )
+        {
+            cursor.done();
+        }
     };
+
+    default void query( IndexCursorProgressor.NodeValueCursor cursor, IndexQuery... query )
+    {
+        try
+        {
+            cursor.initialize( new NodeValueIndexProgressor( query( query ), cursor ), null );
```

Comment:
Why do we use `null` for property keys here? I know that `NodeValueIndexProgressor` doesn't serve any values but that should not be the concern of this default implementation. Through `query` we have access to the property key ids, let's use them.

Using `NodeValueIndexCursor.numberOfProperties()` to determine if values are possible to read directly from index does not work because it depend on what capability the underlying index implementation has and what values are stored. The capabilities for string may not be the same as for number for example. Therefore we can't know if values are available in index or not when we initialize `NodeValueCursor`. I'm working on extending the API for this purpose, so no work need to be done to accommodate for that in this PR. Just wanted to give some context about above comment.